client-core/src/mnemonic.rs

```diff
@@ -38,8 +38,19 @@ impl Mnemonic {
     /// Generate and returns mnemonic words
     #[allow(clippy::new_without_default)]
     #[inline]
-    pub fn new() -> Self {
-        let mnemonic = bip39::Mnemonic::new(MNEMONIC_TYPE, MNEMONIC_LANGUAGE);
+    pub fn new(mnemonics_word_count: u32) -> Self {
+        let flag = if 12 == mnemonics_word_count {
+            MnemonicType::Words12
+        } else if 15 == mnemonics_word_count {
+            MnemonicType::Words15
+        } else if 18 == mnemonics_word_count {
+            MnemonicType::Words18
+        } else if 21 == mnemonics_word_count {
+            MnemonicType::Words21
+        } else {
+            MNEMONIC_TYPE
```

Comment:
Not sure if it's good to silently put the default -- maybe it'll be good to at least display some info/warning message ("Invalid mnemonic word count type, using the default 24 words")?

Reply:
i'll change to
12,15,18,21,24 count only
otherwise display error message, and not continue